start_control_ssm.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register with a hybrid activation
+  activation_code=$(fetch_from_json ".ssm.activation-code" "${USER_DATA}")
+  activation_id=$(fetch_from_json ".ssm.activation-id" "${USER_DATA}")
+  region=$(fetch_from_json ".ssm.region" "${USER_DATA}")
+
+  # Register with AWS Systems Manager (SSM)
+  amazon-ssm-agent -register -code "${activation_code}" -id "${activation_id}" -region "${region}"
+  if [[ "$?" -ne 0 ]]; then
+    last_ssm_agent_error="$(tail -n 1 "/var/log/amazon/ssm/errors.log")"
+    log "Failed to register with AWS Systems Manager (SSM): ${last_ssm_agent_error}"
+    exit 1
+  fi
+}
+
+# Fetch the values from json, and exit on failure (if any)
+fetch_from_json() {
+  local key="${1:?}"
+  local file="${2:?}"
+  local value
+  if ! value=$(jq -e -r '$key' "${file}") || [[ -z "${value}" ]]; then
```

Comment:
Hmm - it seems like it should be `"$key"` rather than `'$key'`?  The shell wouldn't replace `$key` inside single quotes, does jq do that?!  (Even if it does, I think it's unexpected when reading shell, so I'd use double...)

Reply:
It should be "${key}" instead of '$key' since we removed the --arg . Sorry about that.